### ISSUE TITLE
Issues with next version of ggplot2

### ISSUE THREAD

**Issue by thomasp85** [philips-software/latrend#124]
Hi

We preparing the next release of ggplot2 and our reverse dependency checks show that your package is failing with the new version. Looking into it it appears the failure is due to deprecation warnings in ggplot2 not expected in your unit tests.

You can install the release candidate of ggplot2 using `devtools::install_github('tidyverse/ggplot2@v3.4.0-rc')` to test this out.

We plan to submit ggplot2 by the end of October and hope you can have a fix ready before then

Kind regards
Thomas

**Comment by niekdt** [comment]
Thanks for the heads-up Thomas. In the commit above I thought I had correctly replaced the functionality of `aes_string`, however,  I'm struggling of reproducing the full functionality of `aes_string` in a convenient way.

The issue I'm having is that with `aes_string`, it was possible for users to provide either column names, expression strings, or expressions. I cannot find an example anywhere on how to handle these cases in a generalized way.

Consider the following plot:
```R
data = data.frame(Time = 1:6, Y = 6:1)
ggplot(data, aes(x = exp(Y), y = Time)) +
  geom_line()
```

Now if we want to make this generic/dynamic, I previously could just do:
```R
# function inputs
response = quote(exp(Y))
time = "Time"
# dynamic plot
ggplot(data, aes_string(x = response, y = time)) +
  geom_line()
```

What is the tidyverse-compliant way of achieving this?

### Current work-around
In the meantime, I've written the following helper function to retain the desired functionality:
```R
as_language = function(x) {
  if (is.language(x)) {
    x
  } else if (is.character(x)) {
    as.symbol(x)
  } else {
    stop('cannot cast to language')
  }
}
```

```R
map = aes(x = !!as_language(response), y = !!as_language(time))
ggplot(data, mapping = map) +
  geom_line()
```

**Comment by niekdt** [comment]
The fixed version (1.4.3) is now published on CRAN
